web/src/components/tasks/business-formation/business/IsVeteranNonprofit.tsx

```diff
@@ -1,72 +1,27 @@
 import { useConfig } from "@/lib/data-hooks/useConfig";
-import { useFormationErrors } from "@/lib/data-hooks/useFormationErrors";
-import { camelCaseToSentence } from "@/lib/utils/cases-helpers";
-import { FormControl, FormControlLabel, FormHelperText, Radio, RadioGroup } from "@mui/material";
 import { ReactElement, useContext } from "react";
 
 export const IsVeteranNonprofit = (): ReactElement => {
   const { Config } = useConfig();
-  const { state, setFormationFormData } = useContext(BusinessFormationContext);
+  const { state, setFormationFormData, setFieldsInteracted } = useContext(BusinessFormationContext);
   const fieldName = "isVeteranNonprofit";
-  const { doesFieldHaveError } = useFormationErrors();
-  const hasError = doesFieldHaveError(fieldName);
 
   return (
-    <WithErrorBar hasError={hasError} type="ALWAYS" className="margin-top-2">
-      <FormationField fieldName={fieldName}>
-        <FormControl variant="outlined" fullWidth>
-          <label className="margin-right-3 text-bold">
-            <Content>{Config.formation.fields.isVeteranNonprofit.description}</Content>
-          </label>
-          <RadioGroup
-            aria-label={camelCaseToSentence(fieldName)}
-            className="fac"
-            value={state.formationFormData.isVeteranNonprofit ?? ""}
-            onChange={(e): void =>
-              setFormationFormData((previousState) => {
-                return {
-                  ...previousState,
-                  isVeteranNonprofit: JSON.parse(e.target.value),
-                };
-              })
-            }
-            row
-          >
-            <FormControlLabel
-              className="margin-right-0"
-              labelPlacement="end"
-              data-testid={"is-veteran-nonprofit-yes"}
-              value={true}
-              control={<Radio color="primary" />}
-              label={
-                <div className="padding-y-1 margin-right-2">
-                  {Config.formation.fields.isVeteranNonprofit.radioYesText}
-                </div>
-              }
-            />
-            <FormControlLabel
-              labelPlacement="end"
-              data-testid={"is-veteran-nonprofit-no"}
-              value={false}
-              control={<Radio color="primary" />}
-              label={
-                <div className="padding-y-1 margin-right-2">
-                  {Config.formation.fields.isVeteranNonprofit.radioNoText}
-                </div>
-              }
-            />
-          </RadioGroup>
-          {hasError && (
-            <FormHelperText className={"text-error-dark"}>
-              {Config.formation.fields.isVeteranNonprofit.error}
-            </FormHelperText>
-          )}
-        </FormControl>
-      </FormationField>
-    </WithErrorBar>
+    <FormationRadio
+      fieldName={fieldName}
+      title={Config.formation.fields.isVeteranNonprofit.description}
+      values={["true", "false"]}
+      onChange={(e): void => {
```

Comment:
[sand] It looks like almost all implementations of `<FormationRadio>` have the same `onChange` function. Thoughts on making this the default and optionally passing a function to override?

Reply:
the exception is in `NonprofitProvisions`

Every usage uses true/false and the same onChange function, which JSON-parses the "true" and "false" strings into boolean values that get saved.

Whereas the one usage in NonprofitProvisions uses "IN_FORM" / "IN_BYLAWS" which is what necessitates passing in the values, and is also different in the onChange because it won't work if it does the JSON parse.

The thing is that if we did a default, it'd have to go hand-in-hand with "true"/"false" as a default values array, because the two go hand-in-hand.

But I would be worried about the overall potential of someone using the component, and overriding the default values with "something"/"custom" and not realizing it's ALSO necessary to override the onChange as well. I'd want some way to tie them together or something